controller/src/main/java/io/pravega/controller/store/stream/records/StreamSegmentRecord.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream.records;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.pravega.common.ObjectBuilder;
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import io.pravega.shared.segment.StreamSegmentNameUtils;
+import lombok.Builder;
+import lombok.Data;
+
+import java.io.IOException;
+import java.util.AbstractMap;
+
+@Data
+@Builder
+/**
+ * Data class for Stream segment record.
+ */
+public class StreamSegmentRecord {
+    public static final StreamSegmentRecordSerializer SERIALIZER = new StreamSegmentRecordSerializer();
+
+    private final int segmentNumber;
+    private final int creationEpoch;
+    private final long creationTime;
+    private final double keyStart;
+    private final double keyEnd;
+
+    public static class StreamSegmentRecordBuilder implements ObjectBuilder<StreamSegmentRecord> {
+
+    }
+
+    public long segmentId() {
+        return StreamSegmentNameUtils.computeSegmentId(segmentNumber, creationEpoch);
+    }
+
+    public boolean overlaps(final StreamSegmentRecord segment) {
```

Comment:
It would be very nice if all these helper classes contain Javadoc for all publicly visible methods. It's very hard for someone else to understand what the method is supposed to do by just looking at the name and code.